internal/config/reminder/config.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/stacklok/minder/internal/config"
+)
+
+// Config contains the configuration for the reminder service
+type Config struct {
+	Database         config.DatabaseConfig `mapstructure:"database"`
+	RecurrenceConfig RecurrenceConfig      `mapstructure:"recurrence"`
+	EventConfig      EventConfig           `mapstructure:"events"`
+	CursorFile       string                `mapstructure:"cursor_file"`
+	LoggingLevel     string                `mapstructure:"logging_level"`
+}
+
+// RecurrenceConfig contains the configuration for the reminder recurrence
+type RecurrenceConfig struct {
+	// Interval is the time between reminders
+	Interval string `mapstructure:"interval"`
+	// BatchSize is the number of reminders to process at once. Batch size cannot be less than
+	// MaxPerProject * MinProjectFetchLimit.
+	BatchSize int `mapstructure:"batch_size"`
+	// MaxPerProject is the maximum number of reminders per project in a batch
+	MaxPerProject int `mapstructure:"max_per_project"`
+	// MinProjectFetchLimit is the minimum number of projects to fetch in an iteration. Additional
+	// projects are fetched if there is still space in the batch.
+	MinProjectFetchLimit int `mapstructure:"min_project_fetch_limit"`
+	// MinElapsed is the minimum time after last update before sending a reminder
+	MinElapsed string `mapstructure:"min_elapsed"`
+}
+
+// ValidateConfig validates the configuration
+func ValidateConfig(cfg *Config) error {
+	if cfg.RecurrenceConfig.BatchSize <
+		cfg.RecurrenceConfig.MaxPerProject*cfg.RecurrenceConfig.MinProjectFetchLimit {
+		return fmt.Errorf("batch_size %d cannot be less than max_per_project(%d)*min_project_fetch_limit(%d)=%d",
+			cfg.RecurrenceConfig.BatchSize,
+			cfg.RecurrenceConfig.MaxPerProject,
+			cfg.RecurrenceConfig.MinProjectFetchLimit,
+			cfg.RecurrenceConfig.MaxPerProject*cfg.RecurrenceConfig.MinProjectFetchLimit)
+	}
+
+	minElapsed := cfg.RecurrenceConfig.MinElapsed
+	if _, err := time.ParseDuration(minElapsed); err != nil {
+		return fmt.Errorf("invalid min_elapsed: %s", minElapsed)
+	}
+
+	interval := cfg.RecurrenceConfig.Interval
+	if _, err := time.ParseDuration(interval); err != nil {
+		return fmt.Errorf("invalid interval: %s", interval)
+	}
+
+	if cfg.EventConfig == (EventConfig{}) {
+		return fmt.Errorf("event config is empty, required for sending events to minder server")
+	}
+
+	return nil
+}
+
+// RegisterReminderFlags registers the flags for the minder cli
+func RegisterReminderFlags(v *viper.Viper, flags *pflag.FlagSet) error {
+	if err := config.BindConfigFlag(v, flags, "cursor_file", "cursor-file",
+		"/tmp/reminder-cursor", "DB Cursor file path for reminder", flags.String); err != nil {
+		return err
+	}
+
+	if err := config.BindConfigFlag(v, flags, "logging_level", "logging-level",
+		"debug", "Logging level for reminder", flags.String); err != nil {
+		return err
+	}
+
+	if err := config.RegisterDatabaseFlags(v, flags); err != nil {
+		return err
+	}
+
+	return registerRecurrenceFlags(v, flags)
+}
```

Comment:
Am I missing something here? `setViperStructDefaults` does not do anything with flags. It simply sets viper defaults. We would still have to bind flags and provide them a default value. However, that default value we want to attach with flags can be queried from viper, so that is the only duplication reduction I see here.